janitor/functions/conditional_join.py

```diff
@@ -956,92 +969,104 @@ def _create_frame(
     if not df.columns.intersection(right.columns).empty:
         df, right = _create_multiindex_column(df, right)
 
-    if indicator:
+    def _add_indicator(indicator, how, index_size, nlevels, columns):
```

Comment:
It looks like we're missing a docstring for this inner function, which might be helpful for the future.

I've asked ChatGPT for a docstring, and it looks correct. Can you double-check, @samukweku?

```python
def _add_indicator(
    indicator: Union[bool, str],
    how: str,
    index_size: int,
    nlevels: int,
    columns: pd.Index,
) -> Tuple[Union[str, Tuple[str, ...]], pd.Categorical]:
    """Add an indicator column to the DataFrame.

    Args:
        indicator: Indicator column name or True for default name "_merge".
        how: Type of join operation ("inner", "left", "right").
        index_size: Size of the index on which the indicator is based.
        nlevels: Number of levels in the DataFrame's column MultiIndex.
        columns: Union of the columns from both DataFrames.

    Returns:
        A tuple containing the indicator column name and a Categorical array
        representing the indicator values for each row.

    Raises:
        ValueError: If the indicator name is already present in the columns.

    """
    mapping = {"left": "left_only", "right": "right_only", "inner": "both"}
    categories = ["left_only", "right_only", "both"]

    if isinstance(indicator, bool):
        indicator = "_merge"

    if indicator in columns:
        raise ValueError(
            "Cannot use the name of an existing column for the indicator column."
        )

    if nlevels > 1:
        indicator = (indicator,) + ("",) * (nlevels - 1)

    if not index_size:
        arr = pd.Categorical([], categories=categories)
    else:
        arr = pd.Categorical(
            [mapping[how]],
            categories=categories,
        )
        if index_size > 1:
            arr = arr.repeat(index_size)

    return indicator, arr
```